runtime/js/10_permissions.js

```diff
@@ -197,6 +197,19 @@
       return PromiseResolve(cache(desc, state));
     }
 
+    querySync(desc) {
```

Comment:
These seem basically identical to the async versions. Perhaps we could update the async versions to just call the sync versions and have the result in a promise. That would help keep them in sync.

Reply:
Good idea - will do.

Reply:
Done